File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/model/workbook/IWorkBook.kt
package com.opencloudsheet.model.workbook

import com.opencloudsheet.metadata.MetadataManager
import com.opencloudsheet.model.metadata.SheetMetadata
import com.opencloudsheet.model.worksheet.IWorksheetRow
import com.opencloudsheet.model.worksheet.IWorkSheet

interface IWorkBook {
    fun getId(): String
    fun getName(): String
    fun getWorkBookEntry(): MetadataManager.WorkBookEntry

    suspend fun initialize()
    suspend fun <T : IWorksheetRow> createSheet(type: Class<T>, name: String, description: String): IWorkSheet<T>
    suspend fun getSheets(): List<SheetMetadata>
    suspend fun <T : IWorksheetRow> getSheet(name: String, type: Class<T>): IWorkSheet<T>?
    suspend fun deleteSheet(name: String)
}

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Model/Workbook/IWorkBook.swift
//
//  IWorkBook.swift
//  OpenCloudSheet
//
//  Interface for workbook operations supporting multiple sheet types
//

import Foundation

public protocol IWorkBook {
    func getId() -> String
    func getName() -> String
    func getWorkBookEntry() -> WorkBookEntry

    func initialize() async throws
    func createSheet<T: IWorksheetRow>(type: T.Type, name: String, description: String) async throws -> any IWorkSheet<T>
    func getSheets() async throws -> [SheetMetadata]
    func getSheet<T: IWorksheetRow>(name: String, type: T.Type) async throws -> (any IWorkSheet<T>)?
    func deleteSheet(name: String) async throws
}
